api/types.proto

```diff
@@ -458,6 +458,11 @@ enum TaskState {
 	SHUTDOWN = 640 [(gogoproto.enumvalue_customname)="TaskStateShutdown"]; // orchestrator requested shutdown
 	FAILED = 704 [(gogoproto.enumvalue_customname)="TaskStateFailed"]; // task execution failed with error
 	REJECTED = 768 [(gogoproto.enumvalue_customname)="TaskStateRejected"]; // task could not be executed here.
+	// The main purpose of the REMOVE state is to correctly handle service deletions
+	// and scale downs. This allows us to keep track of tasks that have been marked
+	// for deletion, but can't yet be removed because the agent is in the process of
+	// shutting them down.
```

Comment:
We should also mention that tasks marked with this states are removed from the system once they have been shut down by the agent.